src/ol/source/wmtssource.js

```diff
@@ -481,3 +533,125 @@ ol.source.WMTS.optionsFromCapabilities = function(wmtsCap, config) {
+ * @param {ol.proj.Projection} projection Projection.
+ * @param {!Object=} opt_params Optional GetFeatureInfo params.
+ *     `INFOFORMAT` at least should be provided if not provided in the
+ *     getFeatureInfoOptions on create.
+ * @return {string|undefined} GetFeatureInfo URL.
+ * @api
+ */
+ol.source.WMTS.prototype.getGetFeatureInfoUrl =
+    function(coordinate, resolution, projection, opt_params) {
+  if (!opt_params) {
+    opt_params = {};
+  }
+
+  goog.asserts.assert(!('VERSION' in opt_params));
+  goog.asserts.assert(('INFOFORMAT' in opt_params) ||
+      ('infoFormat' in this.getFeatureInfoOptions_));
+
+  var pixelRatio = this.pixelRatio_;
+  if (isNaN(pixelRatio)) {
+    return undefined;
+  }
+
+
+  var tileGrid = /** @type {ol.tilegrid.WMTS} */ (this.tileGrid);
+
+
+  var tileCoord = tileGrid.getTileCoordForCoordAndResolution(
+      coordinate, resolution);
+
+  var tmpExtent = ol.extent.createEmpty();
+  var tmpTileCoord = [0, 0, 0];
+  // TODO: this code is duplicated from createFromWMTSTemplate function
+  var getTransformedTileCoord = function(tileCoord, tileGrid, projection) {
+    goog.asserts.assert(!goog.isNull(tileGrid));
+    if (tileGrid.getResolutions().length <= tileCoord[0]) {
+      return null;
+    }
+    var x = tileCoord[1];
+    var y = -tileCoord[2] - 1;
+    var tileExtent = tileGrid.getTileCoordExtent(tileCoord, tmpExtent);
+    var extent = projection.getExtent();
+
+    if (!goog.isNull(extent) && projection.isGlobal()) {
+      var numCols = Math.ceil(
+          ol.extent.getWidth(extent) /
+          ol.extent.getWidth(tileExtent));
+      x = goog.math.modulo(x, numCols);
+      tmpTileCoord[0] = tileCoord[0];
+      tmpTileCoord[1] = x;
+      tmpTileCoord[2] = tileCoord[2];
+      tileExtent = tileGrid.getTileCoordExtent(tmpTileCoord, tmpExtent);
+    }
+    if (!ol.extent.intersects(tileExtent, extent) ||
+        ol.extent.touches(tileExtent, extent)) {
+      return null;
+    }
+    return ol.tilecoord.createOrUpdate(tileCoord[0], x, y);
+  };
+
+  var tileExtent = tileGrid.getTileCoordExtent(tileCoord);
+  var transformedTileCoord = getTransformedTileCoord(tileCoord,
+      tileGrid, projection);
+
+  if (tileGrid.getResolutions().length <= tileCoord[0]) {
+    return undefined;
+  }
+
+  var tileResolution = tileGrid.getResolution(tileCoord[0]);
+  //var tileExtent = tileGrid.getTileCoordExtent(
+  //    tileCoord, this.tmpExtent_);
+  //var tileSize = tileGrid.getTileSize(tileCoord.z);
+  var tileMatrix = tileGrid.getMatrixIds()[tileCoord[0]];
+
+  var baseParams = {
+    'SERVICE': 'WMTS',
+    'VERSION': '1.0.0',
+    'REQUEST': 'GetFeatureInfo',
+    'LAYER': this.layer_,
+    'TileCol': transformedTileCoord[1],
+    'TileRow': transformedTileCoord[2],
+    'TileMatrix': tileMatrix,
+    'TileMatrixSet': this.matrixSet_
+  };
+
+  goog.object.extend(baseParams, opt_params);
+  if (goog.isDef(this.getFeatureInfoOptions_.infoFormat)) {
+    goog.object.set(baseParams, 'INFOFORMAT',
```

Comment:
we've decided not to use goog.object.set anymore see: https://github.com/openlayers/ol3/blob/master/CONTRIBUTING.md#follow-openlayers-3s-coding-style